app/src/main/java/org/fossasia/openevent/general/event/EventService.kt

```diff
@@ -25,22 +26,29 @@ class EventService(private val eventApi: EventApi, private val eventDao: EventDa
     }
 
     fun getSearchEvents(eventName: String): Single<List<Event>> {
-        return eventApi.searchEvents("name", eventName)
-                .map {
-                    eventDao.insertEvents(it)
-                    it
-                }
+        return Single.zip(
+                eventApi.searchEvents("name", eventName),
+                eventDao.getFavouriteEventIds(),
+                BiFunction<List<Event>, List<Long>, List<Event>> { t1, t2 -> updateFavorites(t1, t2) }
+        )
```

Comment:
Wrong logic

Reply:
why is this wrong?

Reply:
works fine locally 👍

Reply:
We want the ultimate object from db and not the API